runtime.md

```diff
@@ -15,3 +15,27 @@ Runs a process in a container. Can be invoked several times.
 Not sure we need that from oc cli. Process is killed from the outside.
 
 This event needs to be captured by oc to run onstop event handlers.
+
+## Hooks
+Hooks allows one to run code before/after various lifecycle events of the container. The state of the container is passed to the hooks over a well-known fd, so the hooks could get the information they need to do their work.
+
+Hooks are placed under a well known directories under which they are search for and executed. The list of directories that will be looked up are:
```

Comment:
We also need to specify what we will do if a hook returns a non-zero exit status.  Should we stop container execution or continue on?  I think we should stop and propagate the error back up to the caller if a hook fails.